dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+             * @type {string} */
+            const DATA_CONTEXT = "context";
+
+            /**
+             * We only checked once if we are in an MS browser.
+             * @type {boolean}
+             */
+            var isMS = isMicrosoftBrowser();
+
+            /**
+             * We only checked once if we are in a Safari browser.
+             * @type {boolean}
+             */
+            var isSafari = isSafariBrowser();
+
+            // private area - functions
+            /**
+             * Checks if the settings contain the button for the action.
+             * If not, looks for the action button in the DOM, using a generated default.
+             * @param {string} id - the id part to look for if
+             * @param {ElementNames} settings - the settings object to use
+             * @param {string} actionName - the action to look for in the settings
+             */
+            function getActionButton(id, settings, actionName) {
+                /**
+                 * Chosen selector to use for the search for the button.
+                 * @type {string} */
+                var selector = "";
+                if (settings[actionName]) {
+                    selector = settings[actionName];
+                } else {
+                    selector = Multiselect.defaults.actionSelector(id, actionName);
+                }
+                return $(selector);
+            }
+
+            /**
+             * Extracts all supported settings for action buttons from the settings or uses defaults.
+             * @param {string} id - id of the element for the left palette
+             * @param {ElementNames} settings - the settings to look in
+             * @returns {ActionButtons} - the wrapper containing all action buttons
+             */
+            function extractActionButtons(id, settings) {
+                return {
+                    $leftAll: getActionButton(id, settings, "leftAll"),
+                    $rightAll: getActionButton(id, settings, "rightAll"),
+                    $leftSelected: getActionButton(id, settings, "leftSelected"),
+                    $rightSelected: getActionButton(id, settings, "rightSelected"),
+                    $undo: getActionButton(id, settings, "undo"),
+                    $redo: getActionButton(id, settings, "redo"),
+                    $moveUp: getActionButton(id, settings, "moveUp"),
+                    $moveDown: getActionButton(id, settings, "moveDown")
+                };
+            }
+
+            /**
+             * This should ensure the action buttons fulfill some required conditions
+             * (e.g. perhaps only one button per multiselect)
+             * (e.g. perhaps we allow other elements than buttons)
+             * @param {ActionButtons} actions - action buttons to check
+             */
+            function validateActionButtons(actions) {
+                // FIXME: Which are required action buttons? Should we error if we can't init them?
```

Comment:
These are placeholders for functions where you could check if the extracted actions are valid buttons or whatever.